data/transactions/logic/fields.go

```diff
@@ -195,7 +195,10 @@ var txnFieldSpecs = []txnFieldSpec{
 	{Sender, StackBytes, 0, 5, false},
 	{Fee, StackUint64, 0, 5, false},
 	{FirstValid, StackUint64, 0, 0, false},
-	{FirstValidTime, StackUint64, 0, 0, false},
+	// FirstValidTime was originally a dummy field in v0, and enabled in v6.
+	// Setting version to 6 will cause assembling programs with version 0-5 to fail
+	// but it does not matter since FirstValidTime is invalid runtime field on TEAL v0-v5
+	{FirstValidTime, StackUint64, 6, 6, false},
```

Comment:
nit: move this to the end of this slice, so that element here would be grouped by the teal logic ver.

Reply:
This would work since there is some internal logic requires that field name values matched to positions in `txnFieldSpecs`. Added blank lines around to make it visible.